### PR TITLE
feat(torchx): indexed add

### PR DESCRIPTION
**Pull request by polvalente**

### CODE REVIEW

File: nx/lib/nx.ex [hunk 1]
```diff
@@ -4946,9 +4946,9 @@ defmodule Nx do
         [3.0]
       >
 
-      iex> Nx.indexed_add(Nx.tensor([1], type: {:u, 32}), Nx.tensor([[0], [0]]), Nx.tensor([1, 1], type: {:u, 64}))
+      iex> Nx.indexed_add(Nx.tensor([1], type: {:s, 32}), Nx.tensor([[0], [0]]), Nx.tensor([1, 1], type: {:s, 64}))
```
[the remark this answers is not in the document]
Comment (polvalente) [reply nx/lib/nx.ex:4949]: The alternative would be the usual:

copy them to unit tests, change this one and ignore the doctests for indexed_add

[the remark this answers is not in the document]
Comment (josevalim) [reply nx/lib/nx.ex:4949]: @polvalente I think we can fix this in torchx though, no? We may simply need to cast before and after the operation?

[the remark this answers is not in the document]
Comment (polvalente) [reply nx/lib/nx.ex:4949]: Both u32 and u64 aren't supported in Torch :(

[the remark this answers is not in the document]
Comment (josevalim) [reply nx/lib/nx.ex:4949]: Ah, of course!

**Review by josevalim — APPROVED** [review]
Let's add tests for out of band access and ship it!

File: nx/lib/nx.ex [hunk 2]
```diff
@@ -4946,13 +4946,18 @@ defmodule Nx do
...
 
   ### Error cases
+      iex> Nx.indexed_add(Nx.tensor([[1], [2]]), Nx.tensor([[0, 1]]), Nx.tensor([0]))
+      ** (ArgumentError) index 1 is out of bounds for axis 1 in shape {2, 1}
+
+      iex> Nx.indexed_add(Nx.tensor([[1], [2]]), Nx.tensor([[-1, 0]]), Nx.tensor([0]))
+      ** (ArgumentError) index -1 is out of bounds for axis 0 in shape {2, 1}
 
       iex> Nx.indexed_add(Nx.tensor([[1], [2]]), Nx.tensor([[[1, 2, 3]]]), Nx.tensor([0]))
       ** (ArgumentError) indices must be a rank 2 tensor, got: 3
```
Comment (josevalim) [inline nx/lib/nx.ex:4963]:
```suggestion
```

File: nx/lib/nx.ex [hunk 3]
```diff
@@ -8674,6 +8679,12 @@ defmodule Nx do
 
       iex> Nx.gather(Nx.tensor([[1, 2], [3, 4]]), Nx.tensor([[0, 0]], type: {:f, 32}))
       ** (ArgumentError) indices must be an integer tensor, got {:f, 32}
+
+      iex> Nx.gather(Nx.tensor([[1], [2]]), Nx.tensor([[0, 1]]))
+      ** (ArgumentError) index 1 is out of bounds for axis 1 in shape {2, 1}
+
+      iex> Nx.gather(Nx.tensor([[1], [2]]), Nx.tensor([[-1, 0]]))
+      ** (ArgumentError) index -1 is out of bounds for axis 0 in shape {2, 1}
```
Comment (josevalim) [inline nx/lib/nx.ex:8687]:
```suggestion
```

**Review by josevalim — APPROVED** [review]

**Pull request merged**